docker/step/Dockerfile

```diff
@@ -1,23 +1,21 @@
 FROM ubuntu:18.04
 
+# This script assume we are at sqlflow root directory and the directory is already built by sqlflow:dev
 # The default source archive.ubuntu.com is busy and slow. We use the following source makes docker build running faster.
-RUN echo '\n\
-deb http://cn.archive.ubuntu.com/ubuntu/ bionic main restricted universe multiverse \n\
-deb http://cn.archive.ubuntu.com/ubuntu/ bionic-security main restricted universe multiverse \n\
-deb http://cn.archive.ubuntu.com/ubuntu/ bionic-updates main restricted universe multiverse \n\
-deb http://cn.archive.ubuntu.com/ubuntu/ bionic-proposed main restricted universe multiverse \n\
-deb http://cn.archive.ubuntu.com/ubuntu/ bionic-backports main restricted universe multiverse \n\
-' > /etc/apt/sources.list
+COPY docker/dev/find_fastest_resources.sh /usr/local/bin/find_fastest_resources.sh
+COPY docker/step/choose_fastest_apt_pip_sources.sh /choose_fastest_apt_pip_sources.sh
 
-RUN apt-get -qq update && apt-get install -y openjdk-8-jdk maven wget unzip > /dev/null
+RUN /choose_fastest_apt_pip_sources.sh
```

Comment:
It seems that here we can write

```dockerfile
RUN source find_fastest_resources.sh \
  && echo "Choose the fastest APT source ..." \
  && choose_fastest_apt_source \
  && echo "Choose the fastest PIP source ..." \
  && choose_fastest_pip_source
```

And we don't need the new file `choose_fastest_apt_pip_sources.sh`.  Am I correct?

Reply:
Yes, to use `source`, we should use bash and quote the command as a param for -c.
```
RUN /bin/bash `source find_fastest_resources.sh \
  && echo "Choose the fastest APT source ..." \
  && choose_fastest_apt_source \
  && echo "Choose the fastest PIP source ..." \
  && choose_fastest_pip_source'
```